src/python/pants/backend/python/lint/black/rules_integration_test.py

```diff
@@ -90,7 +90,6 @@ def run_black(
     )
     assert len(partitions) == 1
     partition = partitions[0]
-    assert partition.metadata == InterpreterConstraints([expected_ics])
```

Comment:
Okay so partition.metadata is going to have the value from `PythonRuleRunner`.  That is `"['>=3.8,<3.10',]"` in this change and `"['>=3.7,<3.10',]"` before.  In neither case does that match the value in `Black.default_interpreter_constraints[0]` (which uses `CPython<4`) and I am utterly baffled how this ever passed.

This doesn't seem to be a common pattern in other tests.